pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -383,11 +383,29 @@ stdenv.mkDerivation ({
       mv tmp "$d"
     done
 
-    for d in $(grep '^dynamic-library-dirs:' "$packageConfDir"/* | cut -d' ' -f2- | tr ' ' '\n' | sort -u); do
-      for lib in "$d/"*.{dylib,so}; do
-        ln -s "$lib" "$dynamicLinksDir"
+    makeDynamicLinks () {
```

Comment:
@junjihashimoto Is there a reason for putting this in a function?

Also, could you add some more comments to this code?  It is not clear what it is doing or why it is needed.  (I know it didn't have any comments to start out with, but it would be nice to get some comments now, hopefully since you have a good idea of what it is doing.)

Reply:
Ok. Sorry for bothering you.

https://github.com/hasktorch/hasktorch/runs/578941364?check_suite_focus=true
This version using makeDynamicLinks fails.

https://github.com/hasktorch/hasktorch/runs/541868184
The test using `ln -sf` passes.

I seem to misunderstand the problem.